wait-for-deployment

```diff
@@ -36,11 +41,22 @@ get_deployment_jsonpath() {
   kubectl get deployment "${deployment}" -o "jsonpath=${_jsonpath}"
 }
 
-if [[ $# != 1 ]]; then
-  echo "usage: $(basename $0) <deployment>" >&2
+if [[ $# -lt 1 ]]; then
+  echo "usage: $(basename $0) <deployment> [timeout]" >&2
   exit 1
 fi
 
+if [[ $2 -gt 0 ]]; then
+  export timeout=$2
+else 
+  export timeout=$DEFAULT_TIMEOUT
+fi
```

Comment:
Please rewrite the whole block like this:

```bash
declare -xr timeout=${2:-$DEFAULT_TIMEOUT}
if [[ ${timeout} -le 0 ]]; then
  echo "timeout parameter must be greater than zero" >&2
  usage
  exit 1
fi
```

The parameter substitution provides a more concise way to set the variable. Plus, it works even with `nounset` enabled.

`declare -xr` is used to declare the variable as both exported and readonly.

Finally, we want to verify that the timeout value is valid, which is the case _iff_ it is larger than zero. The usage instructions need to be moved into a separate function because we may call it from two separate locations now.